src/metal/rt64_metal.cpp

```diff
@@ -0,0 +1,2419 @@
+        descriptor->setSampleCount(desc.multisampling.sampleCount);
+
+        MTL::TextureUsage usage = metal::mapTextureUsage(desc.flags);
+        // Add shader write usage if this texture might be used as a resolve target
+        if (desc.multisampling.sampleCount == 1) {  // Non-MSAA textures could be resolve targets
+            usage |= MTL::TextureUsageShaderWrite;
+        }
+
+        // Add shader read usage if this texture might be used as a resolve source
+        if (desc.multisampling.sampleCount > 1) { // MSAA textures could be resolve sources
+            usage |= MTL::TextureUsageShaderRead;
+        }
+
+        descriptor->setUsage(usage);
+
+        if (pool != nullptr) {
+            this->mtl = pool->heap->newTexture(descriptor);
+        } else {
+            this->mtl = device->mtl->newTexture(descriptor);
+        }
+
+        // Release resources
+        descriptor->release();
+    }
+
+    MetalTexture::~MetalTexture() {
+        mtl->release();
+    }
+
+    std::unique_ptr<RenderTextureView> MetalTexture::createTextureView(const RenderTextureViewDesc &desc) {
+        return std::make_unique<MetalTextureView>(this, desc);
+    }
+
+    void MetalTexture::setName(const std::string &name) {
+        mtl->setLabel(NS::String::string(name.c_str(), NS::UTF8StringEncoding));
+    }
+
+    // MetalTextureView
+
+    MetalTextureView::MetalTextureView(MetalTexture *texture, const RenderTextureViewDesc &desc) {
+        assert(texture != nullptr);
+        assert(texture->desc.dimension == desc.dimension && "Creating a view with a different dimension is currently not supported.");
+
+        this->texture = texture->mtl->newTextureView(
+            metal::mapPixelFormat(desc.format),
+            texture->mtl->textureType(),
+            { desc.mipSlice, desc.mipLevels },
+            { 0, texture->arrayCount },
+            metal::mapTextureSwizzleChannels(desc.componentMapping)
+        );
+    }
+
+    MetalTextureView::~MetalTextureView() {
+        texture->release();
+    }
+
+    // MetalAccelerationStructure
+
+    MetalAccelerationStructure::MetalAccelerationStructure(MetalDevice *device, const RenderAccelerationStructureDesc &desc) {
+        assert(device != nullptr);
+        assert(desc.buffer.ref != nullptr);
+
+        this->device = device;
+        this->type = desc.type;
+    }
+
+    MetalAccelerationStructure::~MetalAccelerationStructure() {
+        // TODO: Should be handled by ARC
+    }
+
+    // MetalPool
+
+    MetalPool::MetalPool(MetalDevice *device, const RenderPoolDesc &desc) {
+        assert(device != nullptr);
+        this->device = device;
+
+//        MTL::HeapDescriptor *descriptor = MTL::HeapDescriptor::alloc()->init();
+//        // TODO: Set Descriptor properties correctly
+//        descriptor->setType(MTL::HeapTypeAutomatic);
+//
+//        this->heap = device->mtl->newHeap(descriptor);
+//
+//        // Release resources
+//        descriptor->release();
```

Comment:
Cleanup commented code and leave a note indicating that pools are not implemented.